webapp/sources/rudder/rudder-rest/src/main/scala/com/normation/rudder/rest/lift/TechniqueApi.scala

```diff
@@ -886,14 +885,12 @@ class TechniqueAPIService14(
   def getTechniqueJson(editorTechnique: EditorTechnique): IOResult[Json] = {
     import techniqueSerializer.*
     import zio.json.*
-    import zio.json.yaml.DecoderYamlOps
     import TechniqueCompilationIO.codecTechniqueCompilationOutput
     import com.normation.zio.*
-    val outputFile = techniqueCompiler.getCompilationOutputFile(editorTechnique)
-    val json       = (for {
-      content <- IOResult.attempt("error when reading compilation output")(outputFile.contentAsString(StandardCharsets.UTF_8))
-      out     <- content.fromYaml[TechniqueCompilationOutput].toIO
-      json    <- out.toJsonAST.toIO
+    val output = techniqueCompiler.getCompilationOutput(editorTechnique)
```

Comment:
I don't understand why output is declared like that ? Why not in the for ?

Reply:
because... the compiler didn't complain :see_no_evil:

Reply:
:white_check_mark: https://github.com/Normation/rudder/pull/5905/commits/ed28eef83afc9fb4aef84d7d020a68ce36b4d63f